installer/pkg/config-generator/ignition.go

```diff
@@ -13,103 +13,89 @@ import (
-		poolToRole[n] = "etcd"
+
+	ca, err := ioutil.ReadFile(filepath.Join(clusterDir, caPath))
+	if err != nil {
+		return err
 	}
-	return poolToRole
-}
 
-// GenerateIgnConfig generates, if successful, files with the ign config for each role.
-func (c *ConfigGenerator) GenerateIgnConfig(clusterDir string) error {
-	poolToRole := c.poolToRoleMap()
-	for _, p := range c.NodePools {
-		role := poolToRole[p.Name]
-		if _, ok := ignFilesPath[role]; !ok {
-			return fmt.Errorf("unrecognized pool: %s", p.Name)
-		}
+	workerCfg, err := parseIgnFile(workers.IgnitionFile)
+	if err != nil {
+		return fmt.Errorf("failed to parse Ignition config for workers: %v", err)
+	}
 
-		ignCfg, err := parseIgnFile(p.IgnitionFile)
-		if err != nil {
-			return fmt.Errorf("failed to GenerateIgnConfig for pool %s and file %s: %v", p.Name, p.IgnitionFile, err)
-		}
+	// XXX(crawford): The SSH key should only be added to the bootstrap
+	//                node. After that, MCO should be responsible for
+	//                distributing SSH keys.
+	c.embedUserBlock(&workerCfg)
+	c.appendCertificateAuthority(&workerCfg, ca)
+	c.embedAppendBlock(&workerCfg, "worker", "")
 
-		var ignCfgs []ignconfigtypes.Config
-		for i := 0; i < p.Count; i++ {
-			ignCfgs = append(ignCfgs, *ignCfg)
-		}
+	if err = ignCfgToFile(workerCfg, filepath.Join(clusterDir, config.IgnitionPathWorker)); err != nil {
+		return err
+	}
 
-		ca, err := ioutil.ReadFile(filepath.Join(clusterDir, caPath))
-		if err != nil {
-			return err
-		}
+	masterCfg, err := parseIgnFile(masters.IgnitionFile)
+	if err != nil {
+		return fmt.Errorf("failed to parse Ignition config for masters: %v", err)
+	}
 
-		for i := range ignCfgs {
-			c.appendCertificateAuthority(&ignCfgs[i], ca)
-		}
+	for i := 0; i < masters.Count; i++ {
+		ignCfg := masterCfg
 
 		// XXX(crawford): The SSH key should only be added to the bootstrap
 		//                node. After that, MCO should be responsible for
 		//                distributing SSH keys.
-		for i := range ignCfgs {
-			c.embedUserBlock(&ignCfgs[i])
-		}
+		c.embedUserBlock(&ignCfg)
+		c.appendCertificateAuthority(&ignCfg, ca)
+		c.embedAppendBlock(&ignCfg, "master", fmt.Sprintf("etcd_index=%d", i))
 
-		fileTargetPath := filepath.Join(clusterDir, ignFilesPath[role])
-		if role == "master" {
-			for i := range ignCfgs {
-				c.embedAppendBlock(&ignCfgs[i], role, fmt.Sprintf("etcd_index=%d", i))
-				if err = ignCfgToFile(ignCfgs[i], fmt.Sprintf(fileTargetPath, i)); err != nil {
-					return err
-				}
-			}
-		} else {
-			c.embedAppendBlock(&ignCfgs[0], role, "")
-			if err = ignCfgToFile(ignCfgs[0], fileTargetPath); err != nil {
-				return err
-			}
+		if err = ignCfgToFile(ignCfg, filepath.Join(clusterDir, fmt.Sprintf(config.IgnitionPathMaster, i))); err != nil {
+			return err
 		}
 	}
+
 	return nil
 }
 
-func parseIgnFile(filePath string) (*ignconfigtypes.Config, error) {
+func parseIgnFile(filePath string) (ignconfigtypes.Config, error) {
```

Comment:
I'd rather keep the pointer here to avoiding needless copying.  But that's a minor nit compared to this PR as a whole, so I'm fine punting a return to pointers to follow-up work.

Reply:
> ... I'm fine punting a return to pointers to follow-up work.

Filed as #186.